### ISSUE TITLE
Remove Need for Adding Signature in RespondCollationBody

### ISSUE THREAD

**Issue by rauljordan** [prysmaticlabs/geth-sharding#204]
Hi all,

As part of the Ruby Release Product Requirements Doc, we decided that **all** sharding clients should have the ability to respond to requests for collation body data from the shardp2p network. Responding to requests entails the following:

- Receiving a p2p `Message` struct with necessary fields to fetch and construct a collation body from persistent, local storage
- The node receiving the request has to fetch the body information from the persistent shard chain DB
- The node has to construct a full collation instance and send it back to the peer that requested it via shardp2p

Currently, in `sharding/syncer/handlers.go`, `RespondCollationBody` requires the node to add a signature to the collation, as it is not part of the fields received in the p2p.Message request. This **restricts the responding of bodies** to only the proposer that created them, as it is the only node that is able to add the corresponding signature. This issue aims to **change** this feature so that the signature field already comes baked into the p2p.Message request. #196 already added the signature field to the SMC to allow for this. A PR for this should eliminate the following code

```go
header := sharding.NewCollationHeader(msg.ShardID, msg.ChunkRoot, msg.Period, msg.Proposer, nil)
sig, err := signer.Sign(header.Hash())
if err != nil {
	return nil, fmt.Errorf("could not sign received header: %v", err)
}

// Adds the signature to the header before calculating the hash used for db lookups.
header.AddSig(sig)
```

and instead use the signature field from the fetched header. The idea is to not have to sign the header hash and get rid of the code above.

**Comment by prestonvanloon** [comment]
So the task is to remove the code you pasted here? I don't fully understand this.

**Comment by rauljordan** [comment]
@prestonvanloon yes the goal is to use the signature from the p2p.Message request instead of having to compute it on the spot. The current functionality restricts responding to collation body requests only to proposers as they are the only ones that can compute this signature. The goal of a PR for this would be to use the signature from the SMC to do this instead.

**Comment by rauljordan** [comment]
@terenc3t do we still need this?

**Comment by terenc3t** [comment]
This got fixed by #223. `RespondCollationBody` no longer needs a signature

```
func RespondCollationBody(req p2p.Message, collationFetcher types.CollationFetcher) (*messages.CollationBodyResponse, error) {
	// Type assertion helps us catch incorrect data requests.
	msg, ok := req.Data.(messages.CollationBodyRequest)
	if !ok {
		return nil, fmt.Errorf("received incorrect data request type: %v", msg)
	}

	header := types.NewCollationHeader(msg.ShardID, msg.ChunkRoot, msg.Period, msg.Proposer, msg.Signature)

	// Fetch the collation by its header hash from the shardChainDB.
	headerHash := header.Hash()
	collation, err := collationFetcher.CollationByHeaderHash(&headerHash)
	if err != nil {
		return nil, fmt.Errorf("could not fetch collation: %v", err)
	}

	return &messages.CollationBodyResponse{HeaderHash: &headerHash, Body: collation.Body()}, nil
}
```
